optimum/exporters/onnx/model_patcher.py

```diff
@@ -32,7 +32,7 @@
     from transformers.modeling_attn_mask_utils import AttentionMaskConverter
 if is_transformers_version(">=", "4.36"):
     from transformers.modeling_attn_mask_utils import _prepare_4d_causal_attention_mask_for_sdpa
-if is_transformers_version(">=", "4.43"):
+if is_transformers_version(">=", "4.43") and is_transformers_version("<", "4.51.999"):
```

Comment:
this is pretty tricky, it probably requires incrementing the min opset to 14 since the model defaults to sdpa